airbyte-webapp/src/components/ResetDataModal/ResetDataModal.tsx

```diff
@@ -69,10 +73,12 @@ const ResetDataModal: React.FC<IProps> = ({ onClose, onSubmit, modalType }) => {
       <Content>
         {modalText()}
         <ButtonContent>
-          <ButtonWithMargin onClick={onClose} secondary>
+          <ButtonWithMargin onClick={onClose} secondary disabled={isLoading}>
             {modalCancelButtonText()}
           </ButtonWithMargin>
-          <Button onClick={onSubmit}>{modalSubmitButtonText()}</Button>
+          <LoadingButton onClick={onSubmitBtnClick} isLoading={isLoading} disabled={isLoading}>
+            {modalSubmitButtonText()}
```

Comment:
Rather than an invoked function within the return, memoizing this value feels like it would be a better option. Or just not making this a function.